Merge branch 'master' into 1-find_list_of_installed_programs

Shayaomer opened the pull request.

Shayaomer committed:
Merge branch 'master' into 1-find_list_of_installed_programs

Shayaomer committed:
Added conditional create of nvd folder

If nvd doesnt exist, create the folder and download the json files, if it exists do nothing

amitrezilion commented on utils.py:
```diff
@@ -4,6 +4,7 @@
 from os.path import isfile, join
 import zipfile
 import json
```
Rename InstalledSoftware1. what's the 1? also, file names in python should be in snake_case (installed_software)

amitrezilion commented on utils.py:
```diff
@@ -4,6 +4,7 @@
 import json
+import pathlib
 
 
```
InstalledSoftware: remove the main function

Shayaomer replied:
Didn't understand what do you mean by removing the main function,
you meant to make it instead, a function which print the fields of a key ?

amitrezilion commented on utils.py:
```diff
@@ -27,6 +28,10 @@ def getCVE_Dict(year: str) -> {}:
     :return: CVE dictionary
     """
```
InstalledSoftware: method and function name should be camel_case too (`connect_to_mainKey`)

amitrezilion commented on utils.py:
```diff
@@ -27,6 +28,10 @@ def getCVE_Dict(year: str) -> {}:
     :return: CVE dictionary
     """
     cve_dict = {}
```
connect_to_mainKey: reset registry_cursor to None instead of an empty string

amitrezilion commented on utils.py:
```diff
@@ -27,6 +28,10 @@ def getCVE_Dict(year: str) -> {}:
     :return: CVE dictionary
     """
     cve_dict = {}
+    if 'nvd' not in listdir():
```
InstalledSoftware: open_element_by_key - instead of printing, use a logger (python has a built in logger). this way will allow to customize the logging (to a file / print)

amitrezilion commented on utils.py:
```diff
@@ -27,6 +28,10 @@ def getCVE_Dict(year: str) -> {}:
     """
     cve_dict = {}
+    if 'nvd' not in listdir():
+        pathlib.Path('nvd').mkdir(parents=True, exist_ok=True)
```
InstalledSoftware: separate this file into 2 classes - one that queries local registry, one that returns installed programs

amitrezilion commented on utils.py:
```diff
@@ -27,6 +28,10 @@ def getCVE_Dict(year: str) -> {}:
     cve_dict = {}
+    if 'nvd' not in listdir():
+        pathlib.Path('nvd').mkdir(parents=True, exist_ok=True)
+        downloadAllZipsFiles()
```
InstalledSoftware: define consts in the class itself

Shayaomer committed:
some resolves

shimonShouei committed:
logging

Shayaomer committed:
CR resolves

Made a new class of 'installedSoftware' and
connected the new class to the logger.

shimonShouei committed:
Merge branch 'master' into 1-find_list_of_installed_programs
